codeworld-base/src/Internal/CodeWorld.hs

```diff
@@ -57,7 +68,7 @@ simulationOf (initial, step, draw) = do
     rs <- chooseRandoms
     CW.simulationOf (initial rs)
                     (\dt w -> step (w, fromDouble dt))
-                    (toCWPic . draw)
+                    (toCWPic . draw) `catch` reportError
```

Comment:
Please put the catch on the next line (but indented relative to the `do` block).